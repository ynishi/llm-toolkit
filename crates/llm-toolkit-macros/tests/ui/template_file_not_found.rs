// Test that template_file validation catches missing files at compile time

use llm_toolkit::ToPrompt;
use serde::Serialize;

#[derive(Serialize, ToPrompt)]
#[prompt(template_file = "templates/does_not_exist.jinja")]
struct MissingTemplateTest {
    name: String,
}

fn main() {}
