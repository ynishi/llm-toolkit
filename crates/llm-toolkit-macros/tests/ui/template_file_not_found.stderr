error: Template file 'templates/does_not_exist.jinja' not found at compile time.

       Searched in:
         - $CARGO_MANIFEST_DIR/templates/does_not_exist.jinja
         - $PWD/templates/does_not_exist.jinja

       Please ensure:
         1. The template file exists
         2. The path is relative to CARGO_MANIFEST_DIR
         3. There are no typos in the path
 --> tests/ui/template_file_not_found.rs:6:8
  |
6 | struct MissingTemplateTest {
  |        ^^^^^^^^^^^^^^^^^^^
